js/src/features/human-resouce/epics.ts

```diff
@@ -1,58 +1,195 @@
 
 import {
-  fetchUsersVacationsAsync,
-  fetchUserDetailsAsync,
-  fetchProfileAsync,
+  fetchEmployeeAsync,
+  fetchEmployeesAsync,
+  updateEmployeeBirthdayAsync,
+  updateContactsAsync,
+  specifyEmployeeSkillsAsync,
+  addEmployeeNoteAsync,
+  applyEmployeePositionAsync,
+  applyEmployeeSalaryAsync,
 } from "./actions";
 
-export const fetchUsersEpic: RootEpic = (action$, state$, { api }) =>
+// export const fetchUsersEpic: RootEpic = (action$, state$, { api }) =>
+//   action$.pipe(
+//     filter(isActionOf(fetchUsersVacationsAsync.request)),
+//     switchMap((action) =>
+//       from(api.humanResource.fetchVacations()).pipe(
+//         map((result) =>
+//           result.success
+//             ? fetchUsersVacationsAsync.success(result.data)
+//             : fetchUsersVacationsAsync.failure(result.reason),
+//         ),
+//         catchError((message) => of(fetchUsersVacationsAsync.failure(message))),
+//       ),
+//     ),
+//   );
+
+export const fetchEmployeeEpic: RootEpic = (action$, state$, { api }) =>
   action$.pipe(
-    filter(isActionOf(fetchUsersVacationsAsync.request)),
-    switchMap(action =>
-      from(api.humanResource.fetchVacations()).pipe(
-        delay(1000),
-        map(result =>
+    filter(isActionOf(fetchEmployeeAsync.request)),
+    switchMap((action) =>
+      from(api.humanResource.fetchEmployee(action.payload)).pipe(
+        map((result) =>
           result.success
-            ? fetchUsersVacationsAsync.success(result.data)
-            : fetchUsersVacationsAsync.failure(result.reason),
+            ? fetchEmployeeAsync.success(result.data)
+            : fetchEmployeeAsync.failure(result.reason),
         ),
-        catchError(message => of(fetchUsersVacationsAsync.failure(message))),
+        catchError((message) => of(fetchEmployeeAsync.failure(message))),
       ),
     ),
   );
 
-export const fetchUserDetailsEpic: RootEpic = (action$, state$, { api }) =>
+export const fetchEmployeesEpic: RootEpic = (action$, state$, { api }) =>
   action$.pipe(
-    filter(isActionOf(fetchUserDetailsAsync.request)),
-    switchMap(action =>
-      from(api.humanResource.fetchUserDetails(action.payload)).pipe(
-        delay(1000),
-        map(result =>
+    filter(isActionOf([fetchEmployeesAsync.request])),
+    switchMap((action) =>
+      from(api.humanResource.fetchEmployees()).pipe(
+        map((result) =>
           result.success
-            ? fetchUserDetailsAsync.success(result.data)
-            : fetchUserDetailsAsync.failure(result.reason),
+            ? fetchEmployeesAsync.success(result.data)
+            : fetchEmployeesAsync.failure(result.reason),
+        ),
+        catchError((message) => of(fetchEmployeesAsync.failure(message))),
+      ),
+    ),
+  );
+
+export const updateEmployeeBirthdayEpic: RootEpic = (
+  action$,
+  state$,
+  { api },
+) =>
+  action$.pipe(
+    filter(isActionOf(updateEmployeeBirthdayAsync.request)),
+    debounceTime(500),
```

Comment:
Can you clarify why do you need debounce?

Reply:
If you very fast change date :D Yep, it can be deleted

Reply:
Understood. Stay it here